ambry-rest/src/test/java/com.github.ambry.rest/NettyResponseChannelTest.java

```diff
@@ -392,6 +392,23 @@ public void keepAliveTest() {
     channel.close();
   }
 
+  /**
+   * Tests that the underlying network channel is closed when {@link NettyResponseChannel#close()} is called.
+   */
+  @Test
+  public void closeTest() {
+    // request is keep-alive by default.
+    HttpRequest request = createRequestWithHeaders(HttpMethod.GET, TestingUri.Close.toString());
+    EmbeddedChannel channel = createEmbeddedChannel();
+    channel.writeInbound(request);
+
+    // drain the channel of content.
+    while (channel.readOutbound() != null) {
+      ;
```

Comment:
The formatter does this unfortunately. I usually write this as
```
while (channel.readOutbound() != null);
```
but the formatter kicks in and makes it this way